packages/wrangler/src/pages/deployment-tails.tsx

```diff
@@ -0,0 +1,259 @@
+			})
+			// Tail filters;
+			.option("status", {
+				choices: statusChoices,
+				describe: "Filter by invocation status",
+				array: true,
+			})
+			.option("header", {
+				type: "string",
+				requiresArg: true,
+				describe: "Filter by HTTP header",
+			})
+			.option("method", {
+				type: "string",
+				requiresArg: true,
+				describe: "Filter by HTTP method",
+				array: true,
+			})
+			.option("search", {
+				type: "string",
+				requiresArg: true,
+				describe: "Filter by a text match in console.log messages",
+			})
+			.option("sampling-rate", {
+				type: "number",
+				describe: "Adds a percentage of requests to log sampling rate",
+			})
+			.option("ip", {
+				type: "string",
+				requiresArg: true,
+				describe:
+					"Filter by the IP address the request originates from. Use " +
+					'"self" to filter for your own IP',
+				array: true,
+			})
+	);
+}
+
+export async function Handler({
+	deploymentId,
+	projectName,
+	environment,
+	header,
+	ip: clientIp,
+	method,
+	samplingRate,
+	search,
+	status,
+	format = "pretty",
+	debug,
+	...args
+}: Options) {
+	if (status && !isStatusChoiceList(status)) {
+		throw new FatalError(
+			"Invalid value for `--status`. Valid options: " + statusChoices.join(", ")
+		);
+	}
+
+	if (format === "pretty") {
+		await printWranglerBanner();
+	}
+
+	const isInteractive = process.stdin.isTTY;
+	const config = readConfig(args.config as ConfigPath, args);
+	const pagesConfig = getConfigCache<PagesConfigCache>(
+		PAGES_CONFIG_CACHE_FILENAME
+	);
+	const accountId = await requireAuth(pagesConfig);
+
+	projectName = projectName || pagesConfig.project_name;
+
+	if (!deploymentId && !projectName && isInteractive) {
+		projectName = await promptSelectProject({ accountId });
+	}
+
+	if (!deploymentId && !projectName) {
+		throw new FatalError("Must specify a project name or deployment ID.", 1);
+	}
+
+	if (!deploymentId) {
+		logger.log(
+			"No deploymentId specified. Using latest deployment for ",
+			environment,
+			"environment."
+		);
+
+		const deployments: Array<Deployment> = await fetchResult(
+			`/accounts/${accountId}/pages/projects/${projectName}/deployments`
+		);
+
+		const envDeployments = deployments.filter(
+			(d) => d.environment === environment
+		);
+
+		if (envDeployments.length === 0) {
+			throw new FatalError("No deployments for environment: " + environment, 1);
+		}
+
+		const latestDeployment = envDeployments
+			.map((d) => ({ id: d.id, created_on: new Date(d.created_on) }))
+			.sort((a, b) => +b.created_on - +a.created_on)[0];
+
+		deploymentId = latestDeployment.id;
+	}
+
+	if (!deploymentId || !projectName) {
+		throw new FatalError("An unknown error occurred.", 1);
+	}
+
+	logger.log("Running tail on deploymentId:", deploymentId);
+
+	const filters = translateCLICommandToFilterMessage({
+		header,
+		clientIp,
+		method,
+		samplingRate,
+		search,
+		status,
+	});
+
+	await metrics.sendMetricsEvent("begin pages log stream", {
```

Comment:
`B`egin pages...?

Reply:
None of the other event names are capitalized so keeping as is